url0d38a10ee7_diegopliebana_GEBT_tgz-pJ8-ch_idsia_scenarios_test_EvolveWithChangingSeedsJ8/src/ch/idsia/mario/engine/Art.java

```diff
@@ -0,0 +1,91 @@
+package ch.idsia.mario.engine;
+
+import javax.imageio.ImageIO;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+
+
+public class Art
+{
+    public static Image[][] mario;
+    public static Image[][] racoonmario;
+    public static Image[][] smallMario;
+    public static Image[][] fireMario;
+    public static Image[][] enemies;
+    public static Image[][] items;
+    public static Image[][] level;
+    public static Image[][] particles;
+    public static Image[][] font;
+    public static Image[][] bg;
+//    public static Image[][] map;
+//    public static Image[][] endScene;
+//    public static Image[][] gameOver;
+//    public static Image logo;
+//    public static Image titleScreen;
+//    final static String curDir = System.getProperty("user.dir");
+
+    public static void init(GraphicsConfiguration gc)
+    {
+        try
+        {
+            mario = cutImage(gc, "resources/mariosheet.png", 32, 32);
+            racoonmario = cutImage(gc, "resources/racoonmariosheet.png", 32, 32);
+            smallMario = cutImage(gc, "resources/smallmariosheet.png", 16, 16);
+            fireMario = cutImage(gc, "resources/firemariosheet.png", 32, 32);
+            enemies = cutImage(gc, "resources/enemysheet.png", 16, 32);
+            items = cutImage(gc, "resources/itemsheet.png", 16, 16);
+            level = cutImage(gc, "resources/mapsheet.png", 16, 16);
+//            map = cutImage(gc, "resources/worldmap.png", 16, 16);
+            particles = cutImage(gc, "resources/particlesheet.png", 8, 8);
+            bg = cutImage(gc, "resources/bgsheet.png", 32, 32);
+//            logo = getImage(gc, "resources/logo.gif");
+//            titleScreen = getImage(gc, "resources/title.gif");
+            font = cutImage(gc, "resources/font.gif", 8, 8);
+//            endScene = cutImage(gc, "resources/endscene.gif", 96, 96);
+//            gameOver = cutImage(gc, "resources/gameovergost.gif", 96, 64);
+        }
+        catch (Exception e)
+        {
+            e.printStackTrace();
+        }
+
+    }
+
+    private static Image getImage(GraphicsConfiguration gc, String imageName) throws IOException
+    {
+        BufferedImage source = null;
+        try
+        { source = ImageIO.read(Art.class.getResourceAsStream(imageName));        }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *getResourceAsStream*. Currently, there are execution paths that do not contain closure statements, for example, when ImageIO.read throws an exception. Ensure the object returned by *getResourceAsStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).